src/Google.Cloud.Diagnostics.Debug/Google.Cloud.Diagnostics.Debug/LoggingClient.cs

```diff
@@ -0,0 +1,133 @@
+﻿// Copyright 2018 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using Google.Cloud.Logging.V2;
+using StackdriverBreakpoint = Google.Cloud.Debugger.V2.Breakpoint;
+using Google.Cloud.Logging.Type;
+using Google.Api.Gax;
+using Google.Api;
+using System.Linq;
+
+namespace Google.Cloud.Diagnostics.Debug
+{
+    class LoggingClient : ILoggingClient
+    {
+        private static readonly Dictionary<StackdriverBreakpoint.Types.LogLevel, LogSeverity> _logSeverityConversion
+            = new Dictionary<StackdriverBreakpoint.Types.LogLevel, LogSeverity>()
+        {
+                { StackdriverBreakpoint.Types.LogLevel.Info, LogSeverity.Info },
+                {StackdriverBreakpoint.Types.LogLevel.Warning, LogSeverity.Warning },
+                {StackdriverBreakpoint.Types.LogLevel.Error, LogSeverity.Error }
+        };
+
+        private readonly LoggingServiceV2Client _logClient;
+        private readonly AgentOptions _options;
+        private readonly LogName _logName;
+
+        internal LoggingClient(AgentOptions options, LoggingServiceV2Client loggingClient = null)
+        {
+            _logClient = loggingClient ?? LoggingServiceV2Client.Create();
+            _options = GaxPreconditions.CheckNotNull(options, nameof(options));
+            _logName = new LogName(_options.ProjectId, _options.LogName ??
+                $"{_options.Module}-{_options.Version}-debug-log");
+        }
+        
+        /// <summary>
+        /// Substitutes the log message format in breakpoint and writes
+        /// the result as a log entry to the log _logName.
+        /// </summary>
+        /// <returns></returns>
+        public WriteLogEntriesResponse WriteLogEntry(StackdriverBreakpoint breakpoint)
+        {
+            LogEntry logEntry = new LogEntry
+            {
+                LogName = _logName.ToString(),
+                Severity = _logSeverityConversion[breakpoint.LogLevel],
+                TextPayload = SubstituteLogMessageFormat(breakpoint.LogMessageFormat, breakpoint.EvaluatedExpressions.ToList())
+            };
+            MonitoredResource resource = new MonitoredResource { Type = "global" };
```

Comment:
I'm not sure of this but do we always want to write to global even if on GAE or GKE?  Is this what the debugger team said?

Reply:
I don't think there's a guideline on this. But we can try to detect whether we are running on GAE or GKE and uses the corresponding resource.